tools/autograd/gen_autograd_functions.py

```diff
@@ -146,6 +148,11 @@ def save_arg(arg, is_output):
             unpack.append('auto {} = {}_.unpack({});'.format(name, name, ptr))
         elif arg['type'] == 'TensorList':
             saved_variables.append('std::vector<SavedVariable> {}_;'.format(name))
+            release_variables.append('for (auto& sv : {}_)'.format(name))
+            release_variables.append('{')
```

Comment:
I don't really understand the strategy here.  Why are we both:
(1) resetting the data / resetting the grad function and
(2) tracking released on TensorLists and clearing the vector?

Why does both need to happen?

Reply:
Here is explanation from @zou3519: https://github.com/pytorch/pytorch/pull/21533#discussion_r292672784

Reply:
Sorry, I was talking about something else in that comment (brainstorming a different solution that would not use a `released` flag).

@ifedan Just `clear()` (2) is sufficient, I don't think you need to reset the data and grad function (1). `clear()` remove the `SavedVariable`s in the list. Because the SavedVariable owns a tensor and a grad_fn, removing the SavedVariable makes them go away as well.

Reply:
Fixed